kafka/pykafka/broker.py

```diff
@@ -112,3 +116,53 @@ def request_offsets(self, partition_requests):
     def request_metadata(self, topics=[]):
         future = self.handler.request(MetadataRequest(topics=topics))
         return future.get(MetadataResponse)
+
+    ######################
+    #  Commit/Fetch API  #
+    ######################
+
+    def commit_consumer_group_offsets(self, group, topic, partitions, retries):
+        """Commit the offsets of all messages consumed so far by this consumer
+            group with the Offset Commit/Fetch API
+
+        Based on Step 2 here https://cwiki.apache.org/confluence/display/KAFKA/Committing+and+fetching+consumer+offsets+in+Kafka
+        """
+        # XXX how should metadata be handled?
+        preqs = [PartitionOffsetCommitRequest(topic, partition.id,
+                                              partition.offset,
+                                              int(time.time()), '')
+                 for partition in partitions]
+        req = OffsetCommitRequest(self.group, partition_requests=preqs)
+
+        for i in xrange(retries):
+            if i > 0:
+                logger.debug("Retrying in %is" % ((i + 1) ** 2))
+                time.sleep(i ** 2)
+
+            try:
+                self.handler.request(req).get(OffsetCommitResponse)
+            except Exception:
+                logger.debug("Offset commit failed.")
+            else:
+                break
+
+    def fetch_consumer_group_offsets(self, group, topic, partitions, retries):
+        """Fetch the offsets stored in Kafka with the Offset Commit/Fetch API
+
+        Based on Step 2 here https://cwiki.apache.org/confluence/display/KAFKA/Committing+and+fetching+consumer+offsets+in+Kafka
+        """
+        preqs = [PartitionOffsetFetchRequest(topic, partition.id)
```

Comment:
The structure of this function will probably change a bit. I think it will make more sense to have the consumer assemble a bunch of `PartitionOffsetFetchRequest`s and then pass the list into this function.